### PR TITLE
Modernize and enhance 24-hour clock support

### PR DESCRIPTION
**Pull request by penrods**
Several changes to modernize this skill:
* Display of time is now controlled by a Skill Setting
* Add nice_time(), which formats 24-hour time better (e.g. "13 hundred")
* Add query for date or day of week
* Improve pronunciation by changing "timezone" to "time zone"
* Switched from Timer to using a repeating event
* Removed leading '0' from time display for 12-hour clock
* Update license to Apache 2.0, add .gitignore
* Removed unused libraries
* Update README
* Use new naming standards for .voc files

### CODE REVIEW

**Comment by penrods** [comment]
This should also resolve #1 and #2

**Comment by forslund** [comment]
Just started the review something isn't working quite right. When asked about the time, Mycroft answers "It's one thousand one hundred twenty five". (24 hour clock)

File: __init__.py [hunk 1]
```diff
@@ -1,85 +1,128 @@
...
+        use_24hour (bool): output in 24-hour/military or 12-hour format
+        use_ampm (bool): include the am/pm for 12-hour format
+    Returns:
+        (str): The formatted time string
+    '''
 
-LOGGER = getLogger(__name__)
+    if use_24hour:
+        # e.g. "03:01" or "14:22"
+        str = dt.strftime("%H:%M")
```
Comment (forslund) [inline __init__.py]: Not really an issue but consider using `string` instead if `str` since str is the builtin string type.

File: __init__.py [hunk 2]
```diff
@@ -1,85 +1,128 @@
...
+            else:
+                speak += " o " + str[4]  # TODO: Localize
+        else:
+            if str[0] == '0':
+                speak += " " + str[3:5]
+            else:
+                # TODO: convert "23" to "twenty three" in helper method
+
+                # Mimic is speaking "23 34" as "two three 43" :(
+                # but it does say "2343" correctly.  Not ideal for general
```
Comment (forslund) [inline __init__.py]: This is slightly incorrect

**Review by forslund — COMMENTED** [review]
Minor nit-picks

**Pull request merged**